modules/cfe_testcase/src/msg_api_test.c

```diff
@@ -0,0 +1,231 @@
+    CFE_MSG_HeaderVersion_t    hdrVer;
+    CFE_MSG_ApId_t             appId;
+    CFE_MSG_SegmentationFlag_t segFlag;
+    CFE_MSG_SequenceCount_t    seqCnt;
+    CFE_MSG_EDSVersion_t       edsVersion;
+    CFE_MSG_Endian_t           endian;
+    CFE_MSG_PlaybackFlag_t     playbackFlag;
+    CFE_MSG_Subsystem_t        subSystem;
+    CFE_MSG_System_t           system;
+    CFE_MSG_FcnCode_t          fcnCode;
+    CFE_TIME_SysTime_t         msgTime;
+    bool                       hasSecondaryHdr;
+    /* bool                       isValid; */
+
+    memset(&cmd, 0xFF, sizeof(cmd));
+    msgId = 0;
+
+    /* test msg-init */
+    UtAssert_INT32_EQ(CFE_MSG_Init(NULL, CFE_SB_ValueToMsgId(0), sizeof(cmd)), CFE_MSG_BAD_ARGUMENT);
+    UtAssert_INT32_EQ(CFE_MSG_Init(&cmd.Msg, CFE_SB_ValueToMsgId(0), 0), CFE_MSG_BAD_ARGUMENT);
+
+    UtAssert_SUCCESS(CFE_MSG_Init(&cmd.Msg, CFE_SB_ValueToMsgId(msgId), sizeof(cmd)));
+
+    /* test set-msg-size */
+    UtAssert_SUCCESS(CFE_MSG_SetSize(&cmd.Msg, 12u));
+
+    /* test get-msg-size */
+    UtAssert_SUCCESS(CFE_MSG_GetSize(&cmd.Msg, &size));
+    UtAssert_True(size == 12u, "CFE_MSG_SetSize => size = 12u");
+
+    /* test get-type */
+    UtAssert_SUCCESS(CFE_MSG_GetType(&cmd.Msg, &type));
+    /* type has to be ...Type_Cmd because it created within a command */
+    UtAssert_INT32_EQ(type, CFE_MSG_Type_Cmd);
+
+    /* test msg set-type */
+    UtAssert_SUCCESS(CFE_MSG_SetType(&cmd.Msg, CFE_MSG_Type_Tlm));
+    /* check if set-type works like expected */
+    UtAssert_INT32_EQ(CFE_MSG_GetType(&cmd.Msg, &type), CFE_MSG_Type_Tlm);
+
+    /* test get header-version */
+    UtAssert_SUCCESS(CFE_MSG_GetHeaderVersion(&cmd.Msg, &hdrVer));
+    UtAssert_INT32_EQ(CFE_MSG_GetHeaderVersion(&cmd.Msg, NULL), CFE_MSG_BAD_ARGUMENT);
+
+    /* test set header-version */
+    UtAssert_SUCCESS(CFE_MSG_SetHeaderVersion(&cmd.Msg, 0));
+    CFE_MSG_GetHeaderVersion(&cmd.Msg, &hdrVer);
+    UtAssert_True(hdrVer == 0, "hdrVer = 0");
+    UtAssert_INT32_EQ(CFE_MSG_SetHeaderVersion(NULL, hdrVer), CFE_MSG_BAD_ARGUMENT);
+
+    /* test get-has-secondary-header */
+    UtAssert_SUCCESS(CFE_MSG_GetHasSecondaryHeader(&cmd.Msg, &hasSecondaryHdr));
+    UtAssert_INT32_EQ(CFE_MSG_GetHasSecondaryHeader(&cmd.Msg, NULL), CFE_MSG_BAD_ARGUMENT);
+
+    /* test set-has-scondary-header */
+    hasSecondaryHdr = true;
+    UtAssert_SUCCESS(CFE_MSG_SetHasSecondaryHeader(&cmd.Msg, hasSecondaryHdr));
+    hasSecondaryHdr = false;
+    CFE_MSG_GetHasSecondaryHeader(&cmd.Msg, &hasSecondaryHdr);
+    UtAssert_True(hasSecondaryHdr, "CFE_MSG_SetHasSecondaryHeader => CFE_MSG_GetHasSecondaryHeader = true");
+
+    /* test get-apid */
+    UtAssert_SUCCESS(CFE_MSG_GetApId(&cmd.Msg, &appId));
+    UtAssert_INT32_EQ(CFE_MSG_GetApId(&cmd.Msg, NULL), CFE_MSG_BAD_ARGUMENT);
+
+    /* test set-apid */
+    UtAssert_SUCCESS(CFE_MSG_SetApId(&cmd.Msg, 0));
+    UtAssert_True(CFE_MSG_GetApId(&cmd.Msg, &appId) == 0, "CFE_MSG_SetApId => apid = 0");
```

Comment:
This is actually returning CFE_SUCCESS, should check set appId vs returned appId.